### PR TITLE
First draft of a custom Audio Player

### PR DESCRIPTION
**Pull request by muse**
Wrong branch name, correct intentions

### CODE REVIEW

File: src/components/card/Card.vue [hunk 1]
```diff
@@ -14,9 +14,16 @@
     </header>
 
     <section>
-      <h2 class="title" v-if="post_data.type !== 'audio'">{{ post_data.body }}</h2>
-      <audio @click="haltAction" preload="auto" v-else :src="post_data.body" class="audio-post" controls></audio>
-
+      <h2 class="title"
```
Comment (soudy) [inline src/components/card/Card.vue:16]: Can we keep this on one line?

File: src/components/card/Card.vue [hunk 2]
```diff
@@ -14,9 +14,16 @@
...
+        v-if="post_data.type !== 'audio'">
+        {{ post_data.body }}
```
Comment (soudy) [inline src/components/card/Card.vue:18]: Can we use `post` instead of `post_data`?

File: src/components/player/Player.vue.js [hunk 3]
```diff
@@ -0,0 +1,98 @@
+export default {
+  name: 'player',
+  props: {
+    url: {}
+  },
+  mounted() {
+    this.$ = {
+      id: { 'TAudio': 'audio'
+          , 'BPlay': '.button-play'
```
Comment (soudy) [inline src/components/player/Player.vue.js]: Can you format this like it's JavaScript and not Haskell?

File: src/components/player/Player.vue.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
+export default {
+  name: 'player',
+  props: {
+    url: {}
+  },
+  mounted() {
+    this.$ = {
```
Comment (soudy) [inline src/components/player/Player.vue.js]: What does `$` mean here? Using `data()` would be better here.

File: src/components/player/Player.vue.js [hunk 5]
```diff
@@ -0,0 +1,98 @@
...
+          , 'BPlay': '.button-play'
+          , 'BPause': '.button-pause'
+          , 'PBar': '.progress-bar'
+          , 'CTime': '.current-time'
+          , 'MTime': '.max-time'
+      },
+      fn: {
+        timestamp: (number) => {
+          let mod = number % 60
+          return `${Math.floor(number / 60)}:${mod < 10 ? `0${mod}` : mod}`
```
Comment (soudy) [inline src/components/player/Player.vue.js]: Could use some variables here

File: src/components/player/Player.vue.js [hunk 6]
```diff
@@ -0,0 +1,98 @@
...
+  mounted() {
+    this.$ = {
+      id: { 'TAudio': 'audio'
+          , 'BPlay': '.button-play'
+          , 'BPause': '.button-pause'
+          , 'PBar': '.progress-bar'
+          , 'CTime': '.current-time'
+          , 'MTime': '.max-time'
+      },
+      fn: {
```
Comment (soudy) [inline src/components/player/Player.vue.js]: `fn` is non-descriptive

File: src/components/player/Player.vue.js [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+
+      // Determine the relevant audio container and controls by recursively
+      // looking for our parent.
+      const container = (node) => {
+          return node.classList.contains('player')
+            ? node
+            : container(node.parentElement)
+      }
+
+      Object.assign( this.$.el
```
Comment (soudy) [inline src/components/player/Player.vue.js]: Extra space after function call

File: src/components/player/Player.vue.js [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+                   , controls( {}
```
Comment (soudy) [inline src/components/player/Player.vue.js]: Formatting :-1:

File: src/components/player/Player.vue.js [hunk 9]
```diff
@@ -0,0 +1,98 @@
...
+                             , container(event.target)
+                             , this.$.id))
+
+      this.$.el.TAudio.paused
```
Comment (soudy) [inline src/components/player/Player.vue.js]: Put this in a `toggle` function?

File: src/components/player/Player.vue.js [hunk 10]
```diff
@@ -0,0 +1,98 @@
...
+            MTime = Math.floor(this.$.el.TAudio.duration)
+
+      console.log(CTime, MTime)
+
+      this.$.el.CTime.innerHTML = this.$.fn.timestamp(CTime)
+      this.$.el.MTime.innerHTML = this.$.fn.timestamp(MTime)
+      this.$.el.PBar.value      = CTime
+
+      // Move this to a initState.
+      this.$.el.PBar.max        = MTime
```
Comment (soudy) [inline src/components/player/Player.vue.js]: Formatting

**Review by soudy — CHANGES_REQUESTED** [review]

File: src/pages/create-post/CreatePost.vue.js [hunk 11]
```diff
@@ -4,6 +4,7 @@ import Modal from '@/components/modal/Modal'
 import { mapActions } from 'vuex'
 import config from '@/config/config'
 import vSelect from "vue-select"
+//import Player from "@/components/Player/Player"
```
Comment (Steun) [inline src/pages/create-post/CreatePost.vue.js:7]: WTF REMOVE COMMENT RIGHT NAO CODE CLEAN PLS 👎 👎 👎

File: src/components/player/Player.vue.js (hunk 5 above)
Comment (soudy) [reply src/components/player/Player.vue.js]: Also, `mod` is non-descriptive

File: src/components/card/Card.vue [hunk 12]
```diff
@@ -14,9 +14,16 @@
...
+      </h2>
+      <player
+        v-else
+        :url="post_data.body"
+        @click="haltAction"
```
Comment (soudy) [inline src/components/card/Card.vue]: `haltAction` was removed, wasn't it?

File: src/components/player/Player.vue.js [hunk 13]
```diff
@@ -0,0 +1,91 @@
...
+      el: {},
+    }
+  },
+  methods: {
+    _pop(object) {
+      let key = Object.keys(object).pop()
+      return {
+        key: key,
+        value: object[key],
+        remainder: (delete object[key], object)
```
Comment (soudy) [inline src/components/player/Player.vue.js:24]: Isn't it useless to return the remainder, as `delete` mutates the object passed to it?

File: src/components/player/Player.vue.js [hunk 14]
```diff
@@ -0,0 +1,91 @@
...
+      }
+    },
+    _controls(accumulator, container, object) {
+      let item = this._pop(object)
+      accumulator[item.key] = container.querySelector(item.value)
+      return Object.keys(object).length
+        ? this._controls.apply(null, [accumulator, container, item.remainder])
```
Comment (soudy) [inline src/components/player/Player.vue.js:31]: Why apply here and not just `this._controls(accumulator, container, item.remainder)`?

File: src/components/player/Player.vue.js [hunk 15]
```diff
@@ -0,0 +1,91 @@
...
+    },
+    _duration(controls) {
+      return Math.floor(controls.Audio.duration)
+    },
+    _current(controls) {
+      return Math.floor(controls.Audio.currentTime)
+    },
+    _timestamp(number) {
+      let mod = number % 60
+      return `${Math.floor(number / 60)}:${mod < 10 ? `0${mod}` : mod}`
```
Comment (soudy) [inline src/components/player/Player.vue.js:50]: Variables :sob:

File: src/components/player/Player.vue.js [hunk 16]
```diff
@@ -0,0 +1,91 @@
...
+    },
+    _switch(controls) {
```
Comment (soudy) [inline src/components/player/Player.vue.js:52]: `toggle` a better name?

**Pull request closed without merging**